pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -20,24 +20,435 @@
+            target: ─────────────────────────────────╰X──────║───║─┤
+                                                             ╚═══╝
+
+    """
+
+    @staticmethod
+    def _resource_decomp(**kwargs) -> Dict[re.CompressedResourceOp, int]:
+        gate_types = {}
+
+        cnot = re.ResourceCNOT.resource_rep()
+        t = re.ResourceT.resource_rep()
+        h = re.ResourceHadamard.resource_rep()
+        s = re.ResourceS.resource_rep()
+        cz = re.ResourceCZ.resource_rep()
+
+        gate_types[cnot] = 9
+        gate_types[h] = 3
+        gate_types[s] = 1
+        gate_types[cz] = 1
+        gate_types[t] = 2 + 2 * (7)  # T^dagg = 7*T in cost
+
+        return gate_types
+
+    @staticmethod
+    def textbook_resource_decomp() -> Dict[re.CompressedResourceOp, int]:
+        r"""Resources for the Toffoli gate
 
         Resources:
-            The resources come from the following identity expressing Controlled Phase Shift
-            as a product of Phase Shifts and CNOTs.
+            The resources are taken (figure 4.9) from the textbook `Quantum Computation and Quantum Information
+            <https://www.cambridge.org/highereducation/books/quantum-computation-and-quantum-information/01E10196D0A682A6AEFFEA52D53BE9AE#overview>`_.
+
+            The circuit is given by:
+
+            .. code-block:: bash
+
+                0: ───────────╭●───────────╭●────╭●──T──╭●─┤
+                1: ────╭●─────│─────╭●─────│───T─╰X──T†─╰X─┤
+                2: ──H─╰X──T†─╰X──T─╰X──T†─╰X──T──H────────┤
+
+        """
+        gate_types = {}
+
+        cnot = re.ResourceCNOT.resource_rep()
+        t = re.ResourceT.resource_rep()
+        h = re.ResourceHadamard.resource_rep()
+
+        gate_types[cnot] = 6
+        gate_types[h] = 2
+        gate_types[t] = 7
+
+        return gate_types
+
+    def resource_params(self) -> dict:
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> re.CompressedResourceOp:
+        return re.CompressedResourceOp(cls, {})
+
+
+class ResourceMultiControlledX(qml.MultiControlledX, re.ResourceOperator):
+    r"""Resource class for MultiControlledX
+
+    Resources:
+        The resources are obtained from (table 3.) the paper `Polylogarithmic-depth controlled-NOT gates
+        without ancilla qubits <https://www.nature.com/articles/s41467-024-50065-x>`_. Specifically, the
+        resources are given by the following rules:
+
+        * If there is only one control qubit, treat the resources as a :code:`CNOT` gate.
+
+        * If there are two control qubits, treat the resources as a :code:`Toffoli` gate.
+
+        * If there are three control qubits, the resources are two :code:`CNOT` gates and
+          one :code:`Toffoli` gate.
+
+        * If there are more than three control qubits (:math:`n`), the resources are given by
+          :math:`36n - 111` :code:`CNOT` gates.
+
+    """
+
+    # TODO: There is a more efficient resource decomposition, need to update this based on the paper.
```

Comment:
Which paper? You should cite it here.

Reply:
I don't think we should cite the new method until we actually update the implementation to use the new method. For now I think it's fine just to have this reminder so we come back to it at some point 👍🏼